python/tach/parsing/config.py

```diff
@@ -14,7 +14,10 @@ def dump_project_config_to_yaml(config: ProjectConfig) -> str:
     # Instead, should provide custom yaml.Dumper & yaml.Representer or just write our own
     # Sort only constraints and dependencies alphabetically for now
     config.modules.sort(key=lambda mod: mod.path)
```

Comment:
This approach worries me due to the concurrent modification (modifying `config.modules` while iterating over it). I think it would be more natural and more efficient to adjust the sort we are doing on the line before, something like:

```
config.modules.sort(key=lambda mod: (mod.path == ROOT_MODULE_SENTINEL_TAG, mod.path))
```
which would use a tuple as a key to sort the root module to the end (since Python sorts False before True as if they were ints).

Reply:
Gotcha, better approach than concurrent modification of file list while traversing.  Currently used `(bool, module.path)` within config.modules.sort where the only time bool value is True is when its equal to root and all the root paths move to the bottom in a sorted way